filelink.go

```diff
@@ -0,0 +1,60 @@
+package stripe
+
+import (
+	"encoding/json"
+)
+
+// FileLinkParams is the set of parameters that can be used when creating or updating a file link.
+type FileLinkParams struct {
+	Params    `form:"*"`
+	ExpiresAt *int64  `form:"expires_at"`
+	File      *string `form:"file"`
+}
+
+// FileLinkListParams is the set of parameters that can be used when listing file links.
+type FileLinkListParams struct {
+	ListParams   `form:"*"`
+	Created      *int64            `form:"created"`
+	CreatedRange *RangeQueryParams `form:"created"`
+	Expired      *bool             `form:"expired"`
+	File         *string           `form:"file"`
+}
+
+// FileLink is the resource representing a Stripe file link.
+// For more details see https://stripe.com/docs/api#file_links.
+type FileLink struct {
+	Created   int64             `json:"created"`
+	Expired   bool              `json:"expired"`
+	ExpiresAt int64             `json:"expires_at"`
+	File      *FileUpload       `json:"file"`
+	ID        string            `json:"id"`
+	Livemode  bool              `json:"livemode"`
+	Metadata  map[string]string `json:"metadata"`
+	Object    string            `json:"object"`
+	URL       string            `json:"url"`
+}
+
+// UnmarshalJSON handles deserialization of a file link.
+// This custom unmarshaling is needed because the resulting
+// property may be an ID or the full struct if it was expanded.
+func (c *FileLink) UnmarshalJSON(data []byte) error {
+	if id, ok := ParseID(data); ok {
+		c.ID = id
+		return nil
+	}
+
+	type fileLink FileLink
+	var v fileLink
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	*c = FileLink(v)
+	return nil
+}
```

Comment:
Do we need the custom unmarshal here? (i.e., Is file link expanded in any other class?)

Reply:
@brandur-stripe While it's not the case yet it likely will be at some point when we ship more resources. I can imagine this being included in another resource and since I don't think we would have tests for this we would not catch the fact that the deserialization is missing right?

Deferring to you on whether it's better to add it in the future!

Reply:
Actually, `FileLink` is expanded as a list in the `links` attribute of `FileUpload` objects.

Reply:
Okay, fair enough! Do you mind if we add a `filelink_test.go` to test this custom unmarshal logic then? You can copy the tests from dispute almost exactly:

``` go
package stripe

import (
	"encoding/json"
	"testing"

	assert "github.com/stretchr/testify/require"
)

func TestDispute_UnmarshalJSON(t *testing.T) {
	// Unmarshals from a JSON string
	{
		var v Dispute
		err := json.Unmarshal([]byte(`"dp_123"`), &v)
		assert.NoError(t, err)
		assert.Equal(t, "dp_123", v.ID)
	}

	// Unmarshals from a JSON object
	{
		v := Dispute{ID: "dp_123"}
		data, err := json.Marshal(&v)
		assert.NoError(t, err)

		err = json.Unmarshal(data, &v)
		assert.NoError(t, err)
		assert.Equal(t, "dp_123", v.ID)
	}
}
```